### ISSUE TITLE
Length of File

### ISSUE THREAD

**Issue by obstschale** [pcrov/JsonReader#7]
Is it possible to know how large the file is and where the stream is currently? If I know the total size of the file and the position of the stream I could calculate a progress bar or something similar.

**Comment by pcrov** [comment]
The total size of the file you're working on is something you'd need to determine yourself. This information might not be knowable and it'd be a bit out of scope for this lib in any case.

Unfortunately the position is not currently tracked, though I do have ideas along those lines in mind for the next version of the scanner. If it makes it in it'd likely be exposed in the reader as something like the byte offset of the beginning of the node and the node length.

For a rough workaround you can create your own input stream (not a stream in the PHP sense, but a Traversable that spits out a byte at a time) which keeps track of the number of bytes read, and access the info from it directly. Then instead of using e.g. `$reader->open(...)` you'd initialize the reader with ` $reader->init(new Parser(new Lexer($inputStream)))` and go about your business.

This is a "rough" workaround as there is reading ahead that happens (because reasons) so your input stream will be further along than the current node in the reader. Given the accuracy of most progress bars this might be close enough for you.

**Comment by obstschale** [comment]
Thx for your reply. I built a simple work around, which is enough for my simple project. I get the total file size `filesize()` and in my do-while loop I calculate the bits of the current value object.

```
$data  = $reader->value();
$count += strlen( json_encode( $data ) );
```

With `json_encode()` I convert the array to a string to use `strlen()` to find out how large the string is in bytes. I add the output to `$count` to know how much I have processed so far.

This approach is naive and not accurate. The total of `$count` is not the same as `filesize($file)` but that is ok for me. I just need something to show as feedback. My simple console app looks like this now:

![data](https://user-images.githubusercontent.com/1058710/30772290-15ef0fe0-a059-11e7-84bb-a111ecc24adf.gif)

The read value on the lower right corner is the current `$count`. Like I said, it is enough for the user to know that something is happening.
